src/panels/config/energy/components/ha-energy-grid-settings.ts

```diff
@@ -294,13 +294,13 @@ export class EnergyGridSettings extends LitElement {
   }
 
   private _addFromSource() {
+    const gridSource = this.preferences.energy_sources.find(
+      (src) => src.type === "grid"
+    ) as GridSourceTypeEnergyPreference | undefined;
     showEnergySettingsGridFlowFromDialog(this, {
+      current_preference: gridSource,
```

Comment:
I think this naming is a bit confusing, as the `preferences` is more than just the grid sources
```suggestion
      grid_source: gridSource,
```

Reply:
renamed